### PR TITLE
Working version of the court with appeals

### PR DESCRIPTION
**Pull request by clesaege**

### CODE REVIEW

File: decentralized-court/contracts/Court.sol [hunk 1]
```diff
@@ -1,6 +1,10 @@
 pragma solidity ^0.4.6;
 
+import "./Tokens/Token.sol";
+
 // Note, we assume totalSupply < 2^256-1
+// NOTE: All time interactions are for test and demo purpose only.
+// TO DO: Manage the time states of the contract.
```
Comment (n1c01a5) [inline decentralized-court/contracts/Court.sol:7]: maybe `TODO`

File: decentralized-court/contracts/Court.sol [hunk 2]
```diff
@@ -277,9 +264,10 @@ contract Court is Token {
     
     /** Execute the ruling given by the court. 
      *  The ruling must be executable.
+     *  Note that this function can thorow due to the arbitrated contract throwing.
```
Comment (n1c01a5) [inline decentralized-court/contracts/Court.sol:267]: `throw` instead of `thorow`

File: decentralized-court/contracts/Court.sol [hunk 3]
```diff
@@ -396,23 +384,126 @@ contract Court is Token {
...
+            atStake[vote.account]-=vote.stakeA+vote.stakeB; // Remove tokens at Stake such that they can be moved again.
+            balances[vote.account]-=lostStake; // For now just give it to the court as the fee system is not there yet.
+            balances[this]+=lostStake;
             
-        atStake[vote.account]-=vote.stakeA+vote.stakeB; // Remove tokens at Stake such that they can be moved again.
-        balances[vote.account]-=lostStake; // For now just give it to the court as the fee system is not there yet.
-        balances[this]+=lostStake;
-        
-        --dispute.voters.length;// Clean up in order to get some gaz back.
+            --dispute.voters.length;// Clean up in order to get some gaz back.
```
Comment (n1c01a5) [inline decentralized-court/contracts/Court.sol:398]: `gas` instead of `gaz`

File: decentralized-court/contracts/Court.sol [hunk 4]
```diff
@@ -396,23 +384,126 @@ contract Court is Token {
...
+        }
         
         dispute.session=EXECUTABLE; // Now the ruling can be executed.
     }
     
+    /** Make inactive arbitrators loose some tokens.
+     *  TO DO: Send back the caller of the function a few ethers to cover gaz cost.
```
Comment (n1c01a5) [inline decentralized-court/contracts/Court.sol:405]: maybe `TODO`

Comment (n1c01a5) [reply decentralized-court/contracts/Court.sol:405]: `gas` instead of `gaz`

File: decentralized-court/contracts/Court.sol [hunk 5]
```diff
@@ -396,23 +384,126 @@ contract Court is Token {
...
+                throw;
+            uint256 toBeLost=penalty<balances[accounts[i]] ? penalty : balances[accounts[i]]; // Make sure not to loose more than the balance.
+            balances[accounts[i]]-=toBeLost; // Give some of his tokens to the court.
+            balances[this]+=toBeLost;
+            dispute.hasVoted[accounts[i]]=session; // Set him as if he had voted to prevent penalizing him multiple times.
+        }
+    }
+    
+    /** Make inactive jury members loose some tokens.
+     *  TO DO: Send back the caller of the function a few ethers to cover gaz cost.
```
Comment (n1c01a5) [inline decentralized-court/contracts/Court.sol:426]: maybe `TODO`

Comment (n1c01a5) [reply decentralized-court/contracts/Court.sol:426]: `gas`?

File: decentralized-court/contracts/Court.sol [hunk 6]
```diff
@@ -422,27 +513,37 @@ contract Court is Token {
...
+            return disputes[disputeID].voters[appeal][0].stakeB;
+    }
     
-    /** Function the court will call to execute ruling B.
-     *  In most cases, this function should have the modifier onlyCourt.
+    /** Return the list of votes.
+     *  Note that it can throw if disputeID does not exist or appeal is greater than the number of appeals.
+     *  @param disputeID ID of the dispute.
+     *  @param appeal Appeal to return the votes, or 0 to return the vote of the arbitrator.
+     *  @param voteID ID of the vote.
```
Comment (n1c01a5) [inline decentralized-court/contracts/Court.sol:543]: add `*  @param account account  who voted`

**Review by n1c01a5 — CHANGES_REQUESTED** [review]

File: decentralized-court/contracts/Court.sol (hunk 1 above)
Comment (clesaege) [reply decentralized-court/contracts/Court.sol:7]: OK

File: decentralized-court/contracts/Court.sol (hunk 2 above)
Comment (clesaege) [reply decentralized-court/contracts/Court.sol:267]: OK

File: decentralized-court/contracts/Court.sol (hunk 3 above)
Comment (clesaege) [reply decentralized-court/contracts/Court.sol:398]: OK

File: decentralized-court/contracts/Court.sol (hunk 4 above)
Comment (clesaege) [reply decentralized-court/contracts/Court.sol:405]: OK

File: decentralized-court/contracts/Court.sol (hunk 5 above)
Comment (clesaege) [reply decentralized-court/contracts/Court.sol:426]: OK

File: decentralized-court/contracts/Court.sol (hunk 6 above)
Comment (clesaege) [reply decentralized-court/contracts/Court.sol:543]: The param account was not used, so we need to remove it.

**Pull request closed without merging**